023-connected-accounts/README.md

```diff
@@ -0,0 +1,303 @@
+
+Params:
+
+* queries: it should be possible to filter the queries by provider
+
+Response:
+
+List of [Connected Accounts](#connected-account)
+
+##### Get a Connected Account
+
+**GET /v1/users/{userId}/connected/{id}**
+
+Response:
+
+[Connected Account](#connected-account)
+
+##### Refresh Connected Account
+
+**PATCH /v1/users/{userId}/connected/{id}**
+
+Response:
+
+[Connected Account](#connected-account)
+
+##### Delete Connected Account
+
+**DELETE /v1/users/{userId}/connected/{id}**
+
+Response:
+
+None
+
+### Data Structure
+
+<!--
+What kind of changes or additions are required for the Appwrite base collections
+to support this feature. Explain which entities should be added or updated, what new attributes they
+need to have and why. Please think well about the naming conventions and how well they play with other
+Appwrite conventions. Try and stay as consistent with existing patterns as much as possible.
+-->
+
+#### Connected Account
+
+* provider: OAuth2 provider
+* userId: user id from the provider
+* status: connected, disconnected
+* accessToken: access token from the provider
+* accessTokenExpiry: the date when the access token expires in ISO 8601 format
+* providerRefreshToken: refresh token from the provider
+
+### Flow
+
+#### User
+
+##### Connecting to an OAuth2 provider
+
+1. User makes authenticated API call to `GET /v1/account/sessions/oauth2/{provider}`
+2. User is redirected to OAuth2 Provider
+3. User signs in to OAuth2 Provider
+4. User is redirected back to Appwrite at `GET /v1/account/sessions/oauth2/callback/{provider}/{projectId}`
+5. User is redirected to the handler at `GET /v1/account/sessions/oauth2/{provider}/redirect`
+6. Appwrite exchanges the code for a refresh and access token
+7. Appwrite looks up the user Id and email from the provider
+8. Appwrite looks up to see if there is already a Connected Account for the provider
+9. If there is a Connected Account, the access and refresh tokens are refreshed
+10. If there is no Connected Account, a new Connected Account is created 
+11. User is redirected to the success URL
+
+### Key Considerations
+
+1. The email from the OAuth2 provider must not match the email of another user in Appwrite.
+2. The user ID from the OAuth2 provider must not already be connected to another user.
+3. Migration will be required to create the connected accounts collection.
+
+### Supporting Libraries
+
+<!--
+Which different libraries do we need to support the new features?
+Please describe the new library's potential API?
+Avoid using 3rd party libraries when possible, if required - explain why.
+-->
+
+The SDKs will need to be updated to support the new endpoints.
+
+### Breaking Changes
+
+<!--
+Do we break any API or SDK backward compatibility?
+If possible, explain what actions we can take to avoid that.
+-->
+
+The existing OAuth2 APIs will still work as they do now. The only difference is a connected account will be created if one does not exist.
```

Comment:
This is a better place in the RFC to discuss data migrations due to changes in data structure.

Reply:
Added a line.